File: frontend/src/types/report.ts
export interface InterviewReport {
  id: string
  interviewSessionId: string
  reportJson: ReportData
  agentModel: string
  tokensUsed?: number
  processingMs?: number
  createdAt: string
  updatedAt: string
}

export interface ReportData {
  overallScore: number
  dimensionScores: Record<string, number | null>
  questionReviews: QuestionReview[]
  strengths: string[]
  weaknesses: string[]
  hiringRecommendation: HiringRecommendation
  detailedFeedback: string
}

export interface QuestionReview {
  questionText: string
  sourceType: string
  evalMode: string
  score: number
  keyFeedback: string
}

export type HiringRecommendation =
  | 'strongly_recommend'
  | 'recommend'
  | 'consider'
  | 'not_recommend'

export interface InterviewRecord {
  candidateId: string
  candidateName: string
  candidateEmail?: string
  resumeId?: string
  sessionId: string
  interviewDate: string
  overallScore?: number
  recommendation?: string
}

File: backend/app/schemas/report.py
"""Interview report Pydantic schemas."""

from datetime import datetime
from typing import Any, Dict, Optional
from uuid import UUID

from pydantic import BaseModel, Field


class ReportResponse(BaseModel):
    id: UUID
    interview_session_id: UUID
    report_json: Dict[str, Any]
    agent_model: str
    tokens_used: Optional[int] = None
    processing_ms: Optional[int] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class InterviewRecordResponse(BaseModel):
    candidate_id: UUID
    candidate_name: str
    candidate_email: Optional[str] = None
    resume_id: Optional[UUID] = None
    session_id: UUID
    interview_date: datetime
    overall_score: Optional[float] = None
    recommendation: Optional[str] = None
